contracts/drafts/dao/DAO.sol

```diff
@@ -0,0 +1,117 @@
+pragma solidity ^0.5.10;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20Mintable.sol";
+import "../../issuance/IssuanceEth.sol";
+import "../token/ERC20MultiDividendable.sol";
```

Comment:
No, I could not. ERC20DividendableEth supports as `dividendToken` only, well... ethereum. Issuances mint issuanceToken, not ethereum, and that's what gets distributed through the ERC20MultiDividendable pool.

Reply:
Ok, after the VC analogy down below I understand now. 
1. Investors invest ether in the DAO, they get DAO tokens. 
2. The DAO invests ether in ventures, gets venture tokens.
3. The DAO goes and collects ether revenue from the ventures.
4. The ether revenue can be reinvested in ventures by the DAO or withdrawn by investors.
Right?

Reply:
Wrong. Actually, it's right up until point no. 2. Let me complete points 3 and 4:

3. The DAO distributes those venture tokens as dividends to the investors.

4. There is no ether revenue, except:
a. The investors, after dividend disbursement, exchange the venture tokens to ether (UniswapExchange?)
b. The DAO is restarted, so that new investors can board on.

Reply:
I see, I'm thinking on a different kind of investment fund.

If you somehow force the DAO to fund only IssuanceEth.sol ventures that issue ERC20DividendableEth tokens you get a self sustaining entity.

1. The investors fund the DAO.
2. The DAO funds ventures.
3. The DAO gets venture tokens, representing a share on the funded ventures.
4. The DAO collects the ether dividends from the funded ventures.
5. The ether dividends are added to the fundingPool, allowing the DAO to invest in more ventures.

This mechanism removes the need for additional investor rounds. The initial funds raised should be enough to fuel the DAO for perpetuity (in the investment strategies are sound).

The return for the investors could be twofold:
a. The DAO should be an ERC20DividendableEth itself, so that the investors can get some ether as dividends. There should be an action that can be voted to convert funds from the funding pool into dividends to be distributed to investors.
b. If the DAO tokens are dividend-bearing, then the DAO tokens have a market value themselves relative to the size of the funding pool and projects that the DAO has a stake on.

Reply:
So, here's my objection: at point no. 4, you cannot collect ether dividends from the funded ventures. As I said, it makes no sense. You have to collect shares in that venture, not eth. There is no ether-printng venture out there. At some point, ether through the DAO (complicated, collectively) or through individiual action someone has to either: 
1. Trade those venture tokens on an exchange (UniswapExchange?) for etehr (or for something else).
2. Use them on the venture paltform.
3. Burn them.
So, no, no, no. You cannot have `DAO` to be `ERC20DividendableEth`. Even if you would have the `issuanceToken` as `ERC20DividendableEth`, you cannot collect dividends in ether for a token that you bought with ether for that same ether. It's like going into a shop and saying "You can give me this toy if you pay me 10$."

Reply:
> If you somehow force the DAO to fund only IssuanceEth.sol ventures that issue ERC20DividendableEth tokens...

The ventures are IssuanceEth **and** ERC20DividendableEth contracts. They distribute venture tokens (from IssuanceEth) at issuance time and ether dividends (from ERC20DividendableEth) later on.

Ventures take ether from the DAO on issuance. They are supposed to use that ether to make business and obtain even more ether. Then they are supposed to distribute their later ether revenue as dividends back to the DAO. That is how businesses work, really. I give you funds, you make them grow, give me money back.

The DAO collects ether dividends from the ventures in proportion to the venture tokens they got during the issuance of the venture.

Reply:
Hmm, I got your vision, but that would mean:
1. Creating a variation of `IssuanceEth` arround `ERC20DividendableEth`, that instead of storing an address as `issuanceToken` would inherit from `ERC20DividendableEth` and would mint itself to the investors (let's call this variation `IssuanceDividendableEth`.
2. Changing the structure of the DAO.
Do I have green lights for that? Or, perhaps, you see another way forward?

Reply:
1. Exactly. Let's call this contract `Venture.sol` (or `VentureEth.sol` to follow the existing naming scheme).
2. Let's change the structure, it shouldn't be much more complex.

Reply:
1. Exactly. Let's call this contract `Venture.sol` (or `VentureEth.sol` to follow the existing naming scheme).
2. Let's change the structure, it shouldn't be much more complex.

Reply:
Btw, DAO.sol would also inherit from VentureEth.sol. There is some beauty in that.

Reply:
There is. I opened a PR with VentureEth, will continue with the DAO once that one's closed.